proxy/src/console/mgmt.rs

```diff
@@ -44,19 +44,30 @@ pub async fn task_main(listener: TcpListener) -> anyhow::Result<()> {
             .set_nodelay(true)
             .context("failed to set client socket option")?;
 
-        tokio::task::spawn(async move {
-            let span = info_span!("mgmt", peer = %peer_addr);
-            let _enter = span.enter();
+        let span = info_span!("mgmt", peer = %peer_addr);
 
-            info!("started a new console management API thread");
-            scopeguard::defer! {
-                info!("console management API thread is about to finish");
-            }
+        tokio::task::spawn(
+            async move {
+                info!("serving a new console management API connection");
+
+                // these might be long running connections, have a separate logging for cancelling
+                // on shutdown and other ways of stopping.
+                let cancelled = scopeguard::guard(tracing::Span::current(), |span| {
+                    let _e = span.entered();
```

Comment:
Hmm. Why do we need to re-enter the current span?

Reply:
AFAIU it's because we won't have span inside `drop()`, instrument will add it only inside `poll()`.

Reply:
Tested locally, the async block does perform drops before returning the value (which means the drops run during `poll()`). So it isn't strictly necessary

Reply:
Ah good point. This is because the span would had been already dropped if this is a cancellation during shutting down runtime.

Reply:
Sorry, I didn't see above messages when responding.